pkg/kapp/cmd/app/deploy.go

```diff
@@ -172,6 +177,13 @@ func (o *DeployOptions) Run() error {
 		return err
 	}
 
+	if o.PreflightChecks != nil {
```

Comment:
Would we not need a permissions check before we diff? Or do we want to consider permissions needed for the apply?

Reply:
Before apply, we just need `get` permission, if we don't have that, we will fail very early.

Reply:
I think it would be helpful to do it earlier, especially in case someone is trying put together some roles. Having this as a way of debugging roles seems to be pretty cool !

Reply:
Ah, moved it after confirmation before seeing this. I'll move it back :)

Reply:
My main reason for moving it down was to ensure that the `diff` flags work even if we don't have enough permissions. For example, if I am using `--diff-run` along with `--preflight` I think that kapp should exit gracefully after showing the diff (even if we don't have sufficient permissions to make the changes) because we are _not_ making any change.

Reply:
Ah, that makes sense. Thanks @praveenrewar for the extra context! Does it make sense to move this after the diff is shown but before the confirmation step? I'm not really opinionated here because I'm not as well versed in the user flow here as you folks so I'm happy to move it wherever makes the most sense.

Reply:
Fixed linter failures and moved the preflight checks execution to right before the confirmation prompt in the latest push

Reply:
> Does it make sense to move this after the diff is shown but before the confirmation step?

Hmm, I was thinking that a user could still say "no" during the confirmation, but it wouldn’t really be a _preflight_ check if we do it after confirmation, so I think it makes sense.